easybuild/easyblocks/c/comsol.py

```diff
@@ -0,0 +1,133 @@
+##
+# Copyright 2009-2017 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for installing Comsol, implemented as an easyblock
+
+@author: Mikael OEhman (Chalmers University of Technology)
+@author: Ake Sandgren (HPC2N, Umea University)
+"""
+import re
+import os
+import shutil
+import stat
+
+import easybuild.tools.environment as env
+from easybuild.easyblocks.generic.packedbinary import PackedBinary
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import adjust_permissions, copy_file, find_flexlm_license, read_file, write_file
+from easybuild.tools.modules import get_software_root
+from easybuild.tools.run import run_cmd
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+
+class EB_Comsol(PackedBinary):
+    """Support for installing COMSOL."""
+
+    def __init__(self, *args, **kwargs):
+        """Add extra config options specific to COMSOL."""
+        super(EB_Comsol, self).__init__(*args, **kwargs)
+        self.comp_fam = None
+        self.configfile = os.path.join(self.builddir, 'my_setupconfig.ini')
+
+    def configure_step(self):
+        """Configure COMSOL installation: create license file."""
+
+        # The tar file comes from the DVD and has 0444 as permission at
+        # the top dir.
+        adjust_permissions(self.start_dir, stat.S_IWUSR)
+
+        default_lic_env_var = 'LM_LICENSE_FILE'
```

Comment:
This seems to be set by find_flexlm_license automatically so it wouldn't be necessary, and custom_env_vars need not be set to achieve this.
But shouldn't this be set to something COMSOL-specific to allow multiple flex softwares to be installed in one swoop?